spec/03-plugins/42-ai-prompt-guard/01_unit_spec.lua

```diff
@@ -0,0 +1,221 @@
+local PLUGIN_NAME = "ai-prompt-guard"
```

Comment:
Big :+1: for factoring everything into this `handler.execute` function to get superb unit test coverage.

Can you add an integration test to accompany this? It can be minimal given you've covered things well in the unit test, but I'd prefer to have at least _some_ test cases that fire off a real HTTP request.

Reply:
Made an integration suite too.